compiler-builtins/src/probestack.rs

```diff
@@ -131,8 +60,10 @@ macro_rules! define_rust_probestack {
     target_arch = "x86_64",
     not(all(target_env = "sgx", target_vendor = "fortanix"))
 ))]
-core::arch::global_asm!(
-    define_rust_probestack!(
+#[unsafe(naked)]
+#[no_mangle]
+pub unsafe extern "C" fn __rust_probestack() {
```

Comment:
Could you mention in the doc comment that the custom ABI is the reason this is unsafe?

Reply:
Do you mean in the module doc comment? We can add some specific details (about what registers are used etc) if we add a doc comment to the function itself.

Reply:
I was referring to the comments above each of the functions, I guess they're not actually doc comments though they could be. Something like ``(hence the `unsafe`, since the ABI does not actually match `extern "C"`)`` in the paragraph describing ABI is sufficient, unless you want to add more. Just to have a note that it isn't supposed to be `extern "C"`.